backend/doc/migration.md

```diff
@@ -34,7 +34,397 @@ and activate the Tracim virtualenv:
+
+A fix that should work on any database, should be :
+
+```python
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy import Boolean
+
+def rename_boolean_column(
+    table_name: str, old_column_name: str, new_column_name: str, constraint_name: str
+):
+    """
+    Utils to help rename a boolean column supporting:
+    - postgresql
+    - mysql 8.0+
+    - mariadb 10.3+
+    - sqlite
+    """
+    with op.batch_alter_table(table_name) as bop:
+        dialect = op.get_context().dialect
+        if dialect.name == "mysql":
+            if not dialect._is_mariadb and dialect.server_version_info >= (8, 0, 0):
+                bop.drop_constraint(constraint_name, type_="check")
+            bop.alter_column(
+                old_column_name,
+                new_column_name=new_column_name,
+                type_=Boolean(create_constraint=False),
+            )
+            bop.alter_column(new_column_name, type_=Boolean(constraint_name))
+        else:
+            bop.alter_column(old_column_name, new_column_name=new_column_name)
+
+def upgrade():
+    rename_boolean_column("workspaces", "calendar_enabled", "agenda_enabled", "workspaces_chk_2")
+
+
+def downgrade():
+    rename_boolean_column("workspaces", "agenda_enabled", "calendar_enabled", "workspaces_chk_2")
+
+```
+
+As you see, for current alembic version of tracim (1.0.5), you need to:
+- drop explictyl constraint for mysql 8.0+ version
+- do rename for all mysql based database (mariadb included) without creating constraint, then reapply type with
+constraint to be sure constraint will be readded.
+
+## Mysql does already add index to foreign keys
+
+Mysql database automatically add index to foreign keys, this lead to issue with migration downgrade as we cannot drop these index.
+Current way to deal with this, is just ignoring mysql for index add/remove on foreign key as it is really complicated
+to handle this properly with the correct index names.
+
+```python
+from alembic import op
+
+def upgrade():
+    dialect = op.get_context().dialect
+    if dialect.name != "mysql":
+        op.create_index(
+            "idx__content_revisions__content_id", "content_revisions", ["content_id"], unique=False
+        )
+        op.create_index(
+            "idx__content_revisions__workspace_id",
+            "content_revisions",
+            ["workspace_id"],
+            unique=False,
+        )
+
+def downgrade():
+    dialect = op.get_context().dialect
+    if dialect.name != "mysql":
+        op.drop_index("idx__content_revisions__workspace_id", table_name="content_revisions")
+        op.drop_index("idx__content_revisions__content_id", table_name="content_revisions")
+
+```
+
+## Few others useful tricks
+
+### Updating database content in migration revision
+
+There does exist few way to update dabase content in migration revision:
+
+#### Use SQLAlchemy Table
+
+You can use sa table to make updating of database content easier. To do this, you must
```

Comment:
sa table -> `sa.Table`
to make updating of database content easier -> to ease updating the database content